src/main/kotlin/org/rust/ide/actions/RsJoinRawLinesHandler.kt

```diff
@@ -19,9 +19,36 @@ class RsJoinRawLinesHandler : JoinRawLinesHandlerDelegate {
     override fun tryJoinRawLines(document: Document, file: PsiFile, start: Int, end: Int): Int {
         if (file !is RsFile) return CANNOT_JOIN
         if (start == 0) return CANNOT_JOIN
+
+        val joinStructLastLine = tryJoinStructLastLine(file, start)
+        if (joinStructLastLine != CANNOT_JOIN) return joinStructLastLine
+
         return tryJoinSingleExpressionBlock(file, start)
     }
 
+    private fun getCurrentField(file: RsFile, start: Int): PsiElement? {
+        var elem = file.findElementAt(start) ?: return null
+        while (elem != null && elem.elementType != STRUCT_LITERAL_FIELD) elem = elem.prevSibling
+        return elem
+    }
+
+    fun tryJoinStructLastLine(file: RsFile, start: Int): Int {
+        val struct = file.findElementAt(start)?.parentOfType<RsStructLiteral>() ?: return CANNOT_JOIN
+        val lastField = struct.structLiteralBody.structLiteralFieldList.last() ?: return CANNOT_JOIN
+        val currentField = getCurrentField(file, start) ?: return CANNOT_JOIN
+
+        if (currentField != lastField) return CANNOT_JOIN
```

Comment:
Hm, the logic here seems a bit complicated. Perhaps it would be easier to add this to `RsJoinLinesHandler`, which is executed *after* the lines are joined? That way, the condition would be "at the join point, comma is followed by `}`".